src/module.c

```diff
@@ -1379,6 +1348,56 @@ moduleCmdArgAt(const RedisModuleCommandInfoVersion *version,
     return (RedisModuleCommandArg *)((char *)(args) + offset);
 }
 
+/* Helper for categoriesFlagsFromString(). Turns a string representing command
+ * flags into the ACL Category Flags.
+ *
+ * Returns '1' if acl category flag is recognized or
+ * returns '0' if not recognized  */
+int matchAclCategoriesFlags(char *flag, int64_t *acl_categories_flags) {
+    uint64_t this_flag = ACLGetCommandCategoryFlagByName(flag);
+    if (this_flag) {
+        *acl_categories_flags |= (int64_t) this_flag;
+        return 1;
+    }
+    return 0; /* Unrecognized */
+}
+
+/* Helper for RM_SetCommandCategories(). Turns a string representing command
+ * flags into the acl category flags used by Redis ACL which allows users to access 
+ * the module commands by acl categories.
+ * 
+ * It returns the set of acl flags, or -1 if unknown flags are found. */
+int64_t categoriesFlagsFromString(char *aclflags) {
+    int count, j;
+    int64_t acl_categories_flags = 0;
+    sds *tokens = sdssplitlen(aclflags,strlen(aclflags)," ",1,&count);
+    for (j = 0; j < count; j++) {
+        char *t = tokens[j];
+        if (!matchAclCategoriesFlags(t, &acl_categories_flags)) {
+                serverLog(LL_WARNING,"Unrecognized categories flag %s on module load", t);
+                break;
+        }
+    }
+    sdsfreesplitres(tokens,count);
+    if (j != count) return -1; /* Some token not processed correctly. */
+    return acl_categories_flags;
+}
+
+/* RedisModule_SetCommandCategories can be used to set ACL categories to module
```

Comment:
I'm not sure about the name, i think it's missing a reference to the term "ACL"
arguably `RedisModule_SetCommandAclCategories` isn't that much longer...

Reply:
Sure, however I made it RedisModule_SetCommandACLCategories to make it consistent casing with other places we use ACL in module names.